modules/aggregation-layers/src/grid-aggregation-layer.js

```diff
@@ -20,22 +20,315 @@
 
 import AggregationLayer from './aggregation-layer';
 import GPUGridAggregator from './utils/gpu-grid-aggregation/gpu-grid-aggregator';
+import {AGGREGATION_OPERATION} from './utils/aggregation-operation-utils';
+import {Buffer} from '@luma.gl/core';
+import {WebMercatorViewport, log, COORDINATE_SYSTEM} from '@deck.gl/core';
+import GL from '@luma.gl/constants';
+import {Matrix4} from 'math.gl';
+import {getBoundingBox} from './utils/grid-aggregation-utils';
+import {getValueFunc} from './utils/aggregation-operation-utils';
+import BinSorter from './utils/bin-sorter';
+import {pointToDensityGridDataCPU, getGridOffset} from './cpu-grid-layer/grid-aggregator';
 
 export default class GridAggregationLayer extends AggregationLayer {
-  initializeState(aggregationProps) {
+  initializeState({aggregationProps, getCellSize}) {
     const {gl} = this.context;
     super.initializeState(aggregationProps);
     this.setState({
-      gpuGridAggregator: new GPUGridAggregator(gl, {id: `${this.id}-gpu-aggregator`})
+      // CPU aggregation results
+      layerData: {},
+      gpuGridAggregator: new GPUGridAggregator(gl, {id: `${this.id}-gpu-aggregator`}),
+      cpuGridAggregator: pointToDensityGridDataCPU
     });
   }
 
+  updateState(opts) {
```

Comment:
This layer makes a lot of assumptions. You need to document:
- What methods must a child layer implement?
- What states must a child layer populate?
- What shared methods can a child layer use?

Reply:
This layer is internal to the `aggregation-module` and not expected to be used by external users, may be an advanced use case if some one is writing a custom grid aggregation layer. I will write a dev-doc.

Reply:
Added docs for aggregation-layer and grid-aggregation-layer in https://github.com/uber/deck.gl/pull/3954